stdlib/public/runtime/MetadataLookup.cpp

```diff
@@ -732,6 +732,16 @@ swift::_contextDescriptorMatchesMangling(const ContextDescriptor *context,
 static const ContextDescriptor *
 _searchTypeMetadataRecords(TypeMetadataPrivateState &T,
                            Demangle::NodePointer node) {
+#if SWIFT_OBJC_INTEROP
+  // Classes in the __C module are ObjC classes. They never have a
+  // nominal type descriptor, so don't bother to search for one.
+  if (node && node->getKind() == Node::Kind::Class)
```

Comment:
I might have used `&&` a bit more and had fewer `if` statements, but this is fine.

Reply:
My thinking is that we want a separate binding for `child` to make the later conditions more comprehensible, and the check for "does this node have a module name at all?" is conceptually different from "does the module name match the thing we're looking for?" But ultimately the (dis)organization of the code just matches that of the brain it came from.